### PR TITLE
build: Upgrade to React Native 0.73

### PR DESCRIPTION
**Pull request by dcalhoun**
## Related

- https://github.com/WordPress/gutenberg/pull/58475
- https://github.com/wordpress-mobile/gutenberg-mobile/pull/6590

## Description

Incorporate changes from the Gutenberg project to upgrade the React Native dependency to version 0.73.

To test: See https://github.com/WordPress/gutenberg/pull/58475. 

## Regression Notes
1. Potential unintended areas of impact
  Most anything in the editor view. 
1. What I did to test those areas of impact (or what existing automated tests I relied on)
  Manual testing outlined in https://github.com/WordPress/gutenberg/pull/58475. 
1. What automated tests I added (or what prevented me from doing so)
  None, this does not add new functionality. 

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

Testing checklist:
- [ ] WordPress.com sites and self-hosted Jetpack sites.
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by geriux — APPROVED** [review]
LGTM! 🚀 once we start the merging process, let's add an internal note in the release notes about the upgrade.

**Review by dcalhoun — COMMENTED** [review]
All the changes LGTM. 🚀

**Pull request merged**
